extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -166,6 +166,9 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 			if (!pythonExists || forceInstall) {
 				await this.installPythonPackage(backgroundOperation, this._usingExistingPython, this._pythonInstallationPath, this.outputChannel);
 			}
+			// upgrade pip to make sure the right version is installed
+			let cmd = `"${this._pythonExecutable}" -m pip install --upgrade pip`;
```

Comment:
Since this only affects the New Install option, should we skip this upgrade when _usingExistingPython is true? We wouldn't want to force upgrade a user's pip packages as a side effect of another install.

Reply:
Yeah that makes sense.